djangocms_versioning/admin/using_versioning.py

```diff
@@ -0,0 +1,222 @@
+        when a content object is created
+        """
+        super().save_model(request, obj, form, change)
+        if not change:
+            if isinstance(obj, PageContent):
+                # FIXME disabled version creation for `cms.PageContent`
+                # here, as it's already done in `cms.api.create_title`
+                return
+            # create a new version object and save it
+            Version.objects.create(content=obj, created_by=request.user)
+
+    def get_queryset(self, request):
+        """Override manager so records not in published state can be displayed"""
+        from ..helpers import override_default_manager
+
+        with override_default_manager(self.model, self.model._original_manager):
+            queryset = super().get_queryset(request)
+        return queryset
+
+    def get_changelist(self, request, **kwargs):
+        ChangeList = super().get_changelist(request, **kwargs)
+        return versioning_change_list_factory(ChangeList)
+
+    change_form_template = "djangocms_versioning/admin/mixin/change_form.html"
+
+    def render_change_form(
+        self, request, context, add=False, change=False, form_url="", obj=None
+    ):
+        """Add a link to the version table to the change form view"""
+        if "versioning_fallback_change_form_template" not in context:
+            context[
+                "versioning_fallback_change_form_template"
+            ] = super().change_form_template
+
+        return super().render_change_form(request, context, add, change, form_url, obj)
+
+    def get_readonly_fields(self, request, obj=None):
+        """Port permission code from django >= 2.1.
+
+        In later versions of django if a user has view perms but no
+        change perms, fields are set as read-only. This is not the case
+        in django < 2.1.
+        """
+        if obj and not DJANGO_GTE_21:
+            version = Version.objects.get_for_content(obj)
+            if not version.check_modify.as_bool(request.user):
+                if self.fields:
+                    return self.fields
+                if self.fieldsets:
+                    return flatten_fieldsets(self.fieldsets)
+                if self.form.declared_fields:
+                    return self.form.declared_fields
+                return list(
+                    set(
+                        [field.name for field in self.opts.local_fields]
+                        + [field.name for field in self.opts.local_many_to_many]
+                    )
+                )
+        return super().get_readonly_fields(request, obj)
+
+    def has_change_permission(self, request, obj=None):
+        if obj and DJANGO_GTE_21:
+            version = Version.objects.get_for_content(obj)
+            return version.check_modify.as_bool(request.user)
+        return super().has_change_permission(request, obj)
+
+
+class VersioningPageContentAdminMixin(VersioningAdminMixin):
```

Comment:
Part of the filename discussion, I'd see this as being in a file called admin/pageadmin.py as this is an exact comparison to the cms.

Reply:
Called pageadmin.py